### ISSUE TITLE
Publish snapshot maven artifacts

### ISSUE THREAD

**Issue by nirvdrum** [oracle/graal#4666]
**Feature request**

The GraalVM team makes dev releases available, which is generally a fantastic way to try out builds before a release. However, the dev releases frequently do not work with the latest released maven artifacts. That's not terribly surprising, although the error messages when this happens are not particularly helpful. It would be very helpful if snapshot artifacts were published, either to Maven central or a custom repository.

**Is your feature request related to a problem? Please describe.**

I maintain a project called [Native Image Playground](https://github.com/nirvdrum/native-image-playground) where I host examples of using Native Image for embedded use cases. That project uses Maven the _org.graalvm.sdk.graal-sdk_ dependency and the _org.graalvm.buildtools.native-maven-plugin_ plugin. I'm unable to run the project with dev builds without compatible Maven artifacts, which makes it hard to highlight new functionality, test for regressions, and so on.

**Describe the solution you'd like.**

I would like the _org.graalvm.sdk.graal-sdk_ artifact to push snapshot builds somewhere. I think the _org.graalvm.buildtools.native-maven-plugin_ plugin is less prone to breaking with dev GraalVM builds, but if required, I'd like snapshot builds of that as well.

**Describe who do you think will benefit the most.**

Developers using Native Image that want to test their projects ahead of a new release or to verify a bug fix.

**Describe alternatives you've considered.**

The only alternatives I've found are: 1) don't use GraalVM dev builds; or 2) don't use Maven.

**Comment by gilles-duboscq** [comment]
It's a little bit concerning if the GraalVM dev builds don't work when using an older SDK version. The SDK should be a stable API.
Do you have examples of the failures you mention?

Regarding publishing the snapshot artifacts, i think that makes sense. Github packages seem to support maven snpashots.

**Comment by nirvdrum** [comment]
I've had this on my todo list to file. I just went back and checked and the most recent issue I ran into had to do with switch from _org.graalvm.nativeimage.library-support_ to _org.graalvm.sdk.graal-sdk_ artifact. That just happened to coincide with the 22.0.0 -> 22.1.0 upgrade. Changing that artifact now, things work with the dev builds.

If the idea is that the latest published artifacts are always supposed to work with dev builds, that's good enough for me. I'll file issues on any compatibility issues I run into then.
